### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by varunagrawal**
- Automatically install dependencies
- Support versions of CMake<3.12

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,11 @@ else()
   set(_pip_args "--user")
 endif()
 
+# Install dependencies.
+execute_process(COMMAND
```
Comment (ProfFan) [inline CMakeLists.txt:50]: This should be in README not here

**Review by ProfFan — APPROVED** [review]
Great, but we should not put anything with a side effect in CMakeLists

File: CMakeLists.txt (hunk 1 above)
Comment (varunagrawal) [reply CMakeLists.txt:50]: You're right and I know a better way.

**Comment by varunagrawal** [comment]
I used the `setup.py` to enforce dependencies. This should be good to go! Thanks again, @ProfFan.

**Pull request merged**